src/lambda/commands/uploadLambda.ts

```diff
@@ -197,37 +197,41 @@ function createConfirmDeploymentPrompter(lambda: LambdaFunction) {
     })()
 }
 
-interface UploadLambdaWizardState {
+export interface UploadLambdaWizardState {
     readonly uploadType: 'zip' | 'directory'
     readonly targetUri: vscode.Uri
     readonly directoryBuildType: 'zip' | 'sam'
     readonly confirmedDeploy: boolean
     readonly lambda: LambdaFunction
 }
 
-class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
+export class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
     constructor(lambda?: LambdaFunction, invokePath?: vscode.Uri) {
         super({ initState: { lambda } })
         this.form.lambda.region.bindPrompter(() => createRegionPrompter().transform(region => region.id))
 
-        this.form.uploadType.bindPrompter(() => createUploadTypePrompter())
-
-        this.form.targetUri.bindPrompter(({ uploadType }) => {
-            if (uploadType === 'directory') {
-                return createSingleFileDialog({
-                    canSelectFolders: true,
-                    canSelectFiles: false,
-                })
-            } else {
-                return createSingleFileDialog({
-                    canSelectFolders: false,
-                    canSelectFiles: true,
-                    filters: {
-                        'ZIP archive': ['zip'],
-                    },
-                })
-            }
-        })
+        if (invokePath && fs.statSync(invokePath.fsPath).isDirectory()) {
+            this.form.uploadType.setDefault('directory')
+            this.form.targetUri.setDefault(invokePath)
```

Comment:
The steps should be skipped entirely. Does this only set the default?

Reply:
If no prompt is bound then the step is skipped as there's nothing to do. Default value is just to keep the field defined even though nothing was shown. Niche use case but it's the cleanest way.